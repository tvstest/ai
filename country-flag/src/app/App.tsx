import React from 'react'
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom'
import CountrySearch from 'app/pages/CountrySearch'
import CountryList from 'app/pages/CountryList'
import Navbar from 'app/components/Navbar'
import { Routes } from 'app/utils/enums/routes'
import { ThemeProvider } from '@mui/material'
import appTheme from 'app/configs/theme'

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={appTheme}>
        <Navbar />
        <Switch>
          <Route exact path={Routes.Home} component={CountrySearch} />
          <Route exact path={Routes.CountriesByName} component={CountryList} />
          <Redirect to={Routes.Home} />
        </Switch>
      </ThemeProvider>
    </BrowserRouter>
  )
}
export default App
