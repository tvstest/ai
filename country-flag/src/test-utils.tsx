import React from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import Navbar from 'app/components/Navbar'
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom'
import { Routes } from 'app/utils/enums/routes'
import appTheme from 'app/configs/theme'
import CountrySearch from 'app/pages/CountrySearch'
import CountryList from 'app/pages/CountryList'

const AppComponent: React.FC = () => {
  return (
    <ThemeProvider theme={appTheme}>
      <BrowserRouter>
        <Navbar />
        <Switch>
          <Route exact path={Routes.Home} component={CountrySearch} />
          <Route exact path={Routes.CountriesByName} component={CountryList} />
          <Redirect to={Routes.Home} />
        </Switch>
      </BrowserRouter>
    </ThemeProvider>
  )
}

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions
): RenderResult => render(ui, { wrapper: AppComponent, ...options })

export * from '@testing-library/react'
export { customRender as render }
