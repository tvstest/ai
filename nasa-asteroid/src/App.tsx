import { ThemeProvider } from '@mui/material/styles'
import Navbar from 'app/components/Navbar'
import Routes from 'app/components/Routes'
import { Suspense } from 'react'
import theme from './app/config/theme'

const App: React.FC = () => {
  return (
    <Suspense fallback="Loading ...">
      <Navbar />
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </Suspense>
  )
}

export default App
