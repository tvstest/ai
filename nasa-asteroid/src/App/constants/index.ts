export const NASA_BASE_URL = 'https://api.nasa.gov/neo/rest/v1/neo/'
export const NASA_API_KEY_PREFIX = `?api_key=${process.env.REACT_APP_NASA_API_KEY}`
