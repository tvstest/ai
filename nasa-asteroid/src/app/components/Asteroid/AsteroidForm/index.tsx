import { Controller, useForm } from 'react-hook-form'
import { yupResolver } from '@hookform/resolvers/yup'
import * as Yup from 'yup'
import { Button, CircularProgress, Grid, TextField } from '@mui/material'
import { useCallback } from 'react'
import _ from 'lodash'
import { useAsteroidContext } from 'app/context/AsteroidContext'
import {
  loadRandomAsteroidData,
  loadRandomAsteroidId,
  setAsteroidData,
  setAsteroidId,
  setError,
} from 'app/context/AsteroidContext/actions'
import asteroidServices from 'app/services/asteroid-services'
import { IAsteroidForm } from 'app/utility/interfaces/asteroid'

const schema = Yup.object().shape({
  asteroidId: Yup.string()
    .trim()
    .required('Please enter asteroid id')
    .matches(/^[0-9]*$/, 'Asteroid id must be numeric')
    .length(7, 'please enter 7 digit asteroid id'),
})

const defaultValues: IAsteroidForm = {
  asteroidId: '',
}

const AsteroidForm: React.FC<{ handler?: (payload: IAsteroidForm) => void }> =
  ({ handler }) => {
    const { state, dispatch } = useAsteroidContext()

    const loading = state.loadingRandomAsteroidId || state.loadingAsteroidData

    const {
      control,
      handleSubmit,
      reset,
      formState: { errors },
    } = useForm<IAsteroidForm>({
      mode: 'all',
      resolver: yupResolver(schema),
      defaultValues,
    })

    const onSubmit = useCallback(
      async (formData: IAsteroidForm) => {
        handler?.(formData)
        dispatch(setAsteroidId(formData?.asteroidId || ''))

        dispatch(loadRandomAsteroidData())

        try {
          const { data } = await asteroidServices.getAsteroidById(
            formData?.asteroidId || ''
          )
          dispatch(setAsteroidData(data))
        } catch (err) {
          dispatch(
            setError(`Failed to load given asteroid id ${formData?.asteroidId}`)
          )
        }

        reset()
      },
      [dispatch, handler, reset]
    )

    const getRandomAsteroid = useCallback(async () => {
      dispatch(loadRandomAsteroidId())

      try {
        const {
          // eslint-disable-next-line camelcase
          data: { near_earth_objects },
        } = await asteroidServices.getRandomAsteroidId()
        const randomAsteroidData = _.sample(near_earth_objects)

        dispatch(setAsteroidId(randomAsteroidData?.id || ''))

        dispatch(loadRandomAsteroidData())

        const { data } = await asteroidServices.getAsteroidById(
          randomAsteroidData?.id || ''
        )

        dispatch(setAsteroidData(data))
      } catch (e) {
        // eslint-disable-next-line no-console
        console.log(e)
      }
    }, [dispatch])

    return (
      <form noValidate onSubmit={handleSubmit(onSubmit)}>
        <Controller
          name="asteroidId"
          control={control}
          defaultValue=""
          render={({ field }) => (
            <TextField
              {...field}
              required
              fullWidth
              aria-label="Asteroid ID"
              label="Asteroid ID"
              inputProps={{ 'data-testid': 'asteroidId' }}
              role="textbox"
              variant="outlined"
              error={!!errors.asteroidId}
              helperText={errors.asteroidId?.message}
            />
          )}
        />
        <br />
        <br />
        <Grid container spacing={2}>
          <Grid item>
            <Button
              type="submit"
              variant="contained"
              role="button"
              data-testid="form-submit"
              name="submit"
              color="primary"
              disabled={loading}
            >
              Submit
            </Button>
          </Grid>
          <Grid item>
            <Button
              variant="contained"
              color="primary"
              data-testid="random-button"
              onClick={getRandomAsteroid}
              disabled={loading}
              startIcon={loading ? <CircularProgress size={20} /> : undefined}
            >
              Get Random Asteroid
            </Button>
          </Grid>
        </Grid>
      </form>
    )
  }

export default AsteroidForm
