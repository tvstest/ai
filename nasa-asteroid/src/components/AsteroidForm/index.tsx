import { Controller, useForm } from "react-hook-form";
import { IAsteroidForm } from "../../utility/interfaces/asteroid";
import { yupResolver } from "@hookform/resolvers/yup";
import * as Yup from "yup";
import { Button, CircularProgress, Grid, TextField } from "@material-ui/core";
import { useCallback } from "react";
import asteroidServices from "../../services/asteroid-services";
import _ from "lodash";
import { useAsteroidContext } from "../../context/AsteroidContext";
import { ActionType } from "../../context/AsteroidContext/actions";
const schema = Yup.object().shape({
  asteroidId: Yup.string()
    .trim()
    .required("Please enter asteroid id")
    .matches(/^[0-9]*$/, "Asteroid id must be numeric")
    .length(7, "please enter 7 digit asteroid id"),
});

const defaultValues: IAsteroidForm = {
  asteroidId: "",
};

const AsteroidForm: React.FC = () => {
  const { state, dispatch } = useAsteroidContext();

  const {
    control,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<IAsteroidForm>({
    mode: "all",
    resolver: yupResolver(schema),
    defaultValues,
  });

  const onSubmit = useCallback(
    async (formData: IAsteroidForm) => {
      dispatch({
        type: ActionType.SetAsteroidId,
        payload: { asteroidId: formData?.asteroidId || "" },
      });

      dispatch({
        type: ActionType.LoadingAsteroidData,
      });

      try {
        const { data } = await asteroidServices.getAsteroidById(
          formData?.asteroidId || ""
        );
        dispatch({
          type: ActionType.SetAsteroidData,
          payload: data,
        });
      } catch (err) {
        dispatch({
          type: ActionType.SetError,
          payload: {
            message: `Failed to load given asteroid id ${formData?.asteroidId}`,
          },
        });
      }

      reset();
    },
    [dispatch, reset]
  );

  const getRandomAsteroid = useCallback(async () => {
    dispatch({ type: ActionType.LoadingRandomAsteroidId });

    const {
      data: { near_earth_objects },
    } = await asteroidServices.getRandomAsteroidId();
    const randomAsteroidData = _.sample(near_earth_objects);

    dispatch({
      type: ActionType.SetAsteroidId,
      payload: { asteroidId: randomAsteroidData?.id || "" },
    });

    dispatch({
      type: ActionType.LoadingAsteroidData,
    });

    const { data } = await asteroidServices.getAsteroidById(
      randomAsteroidData?.id || ""
    );

    dispatch({
      type: ActionType.SetAsteroidData,
      payload: data,
    });
  }, [dispatch]);

  return (
    <form noValidate onSubmit={handleSubmit(onSubmit)}>
      <Controller
        name="asteroidId"
        control={control}
        defaultValue=""
        render={({ field }) => (
          <TextField
            {...field}
            required
            fullWidth
            label="Asteroid ID"
            variant="outlined"
            error={!!errors.asteroidId}
            helperText={errors.asteroidId?.message}
          />
        )}
      />
      <br />
      <br />
      <Grid container spacing={2}>
        <Grid item>
          <Button
            type="submit"
            variant="contained"
            color="primary"
            disabled={
              state.loadingRandomAsteroidId || state.loadingAsteroidData
            }
          >
            Submit
          </Button>
        </Grid>
        <Grid item>
          <Button
            variant="contained"
            color="primary"
            onClick={getRandomAsteroid}
            disabled={
              state.loadingRandomAsteroidId || state.loadingAsteroidData
            }
            startIcon={
              state.loadingRandomAsteroidId ? (
                <CircularProgress size={20} />
              ) : undefined
            }
          >
            Get Random Asteroid
          </Button>
        </Grid>
      </Grid>
    </form>
  );
};

export default AsteroidForm;
