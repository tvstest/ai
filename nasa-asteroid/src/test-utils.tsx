import { ThemeProvider } from '@mui/material/styles'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import Navbar from 'app/components/Navbar'
import Routes from 'app/components/Routes'
import { BrowserRouter as Router } from 'react-router-dom'
import { Suspense } from 'react'
import theme from './app/config/theme'

const AllTheApp: React.FC = () => {
  return (
    <Suspense fallback="Loading ...">
      <Router>
        <Navbar />
        <ThemeProvider theme={theme}>
          <Routes />
        </ThemeProvider>
      </Router>
    </Suspense>
  )
}

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions
): RenderResult => render(ui, { wrapper: AllTheApp, ...options })

export * from '@testing-library/react'
export { customRender as render }
