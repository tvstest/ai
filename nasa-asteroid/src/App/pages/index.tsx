import { Container, makeStyles } from "@material-ui/core";
import AsteroidDetails from "App/components/AsteroidDetails";
import AsteroidForm from "App/components/AsteroidForm";
import AsteroidContextWrapper from "App/context/AsteroidContext";

const useStyles = makeStyles({
  topMargin: {
    marginTop: 30,
  },
});

const Home = () => {
  const { topMargin } = useStyles();

  return (
    <Container className={topMargin}>
      <AsteroidContextWrapper>
        <AsteroidForm />
        <AsteroidDetails />
      </AsteroidContextWrapper>
    </Container>
  );
};

export default Home;
