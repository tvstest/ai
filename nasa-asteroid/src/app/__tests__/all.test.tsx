import { render, fireEvent, waitFor } from 'test-utils'
import '@testing-library/jest-dom'
import App from 'App'
import AsteroidForm from 'app/components/Asteroid/AsteroidForm'
import { act } from 'react-dom/test-utils'

test('App loads with router and mui theme configuration properly', () => {
  render(<App />)
})

test('Form validations are triggered properly', async () => {
  const { getByTestId, getByText } = render(<AsteroidForm />)
  const inputElement = getByTestId('asteroidId')

  await act(async () => {
    fireEvent.change(inputElement, {
      target: { value: '20019' },
    })
  })

  await act(async () => {
    fireEvent.click(getByTestId('form-submit'))
  })

  await waitFor(async () => {
    expect(getByText('please enter 7 digit asteroid id')).toBeInTheDocument()
  })
})
