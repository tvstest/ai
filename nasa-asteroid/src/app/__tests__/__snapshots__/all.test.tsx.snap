// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the form snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
          >
            NASA
          </h6>
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg makeStyles-topMargin-9 css-1oqqzyl-MuiContainer-root"
      >
        <form
          novalidate=""
        >
          <div
            aria-label="Asteroid ID"
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            role="textbox"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pw3wan-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Asteroid ID
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-sn8fib-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="asteroidId"
                name="asteroidId"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Asteroid ID
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <br />
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1krdwzt-MuiButtonBase-root-MuiButton-root"
                data-testid="form-submit"
                name="submit"
                role="button"
                tabindex="0"
                type="submit"
              >
                Submit
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1krdwzt-MuiButtonBase-root-MuiButton-root"
                data-testid="random-button"
                tabindex="0"
                type="button"
              >
                Get Random Asteroid
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1qp5wg9-MuiTypography-root"
        >
          NASA
        </h6>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg makeStyles-topMargin-9 css-1oqqzyl-MuiContainer-root"
    >
      <form
        novalidate=""
      >
        <div
          aria-label="Asteroid ID"
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          role="textbox"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1pw3wan-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
          >
            Asteroid ID
            <span
              aria-hidden="true"
              class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-sn8fib-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="asteroidId"
              name="asteroidId"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-173wfxe"
              >
                <span>
                  Asteroid ID
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <br />
        <br />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1krdwzt-MuiButtonBase-root-MuiButton-root"
              data-testid="form-submit"
              name="submit"
              role="button"
              tabindex="0"
              type="submit"
            >
              Submit
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1krdwzt-MuiButtonBase-root-MuiButton-root"
              data-testid="random-button"
              tabindex="0"
              type="button"
            >
              Get Random Asteroid
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
